v2/thirdpartyemailpassword/common-customizations/embed-sign-in-up-form.mdx

```diff
@@ -210,53 +493,214 @@ function MyLandingPage() {
+
+<RRDVersionSubTabs>
+
+<TabItem value="v6">
 
 ```tsx
+import React, { useState, useEffect } from "react";
+import Modal from "react-modal";
 import SuperTokens from "supertokens-auth-react";
 import ThirdPartyEmailPassword from "supertokens-auth-react/recipe/thirdpartyemailpassword";
+// highlight-start
+import { SignInAndUp } from "supertokens-auth-react/recipe/thirdpartyemailpassword/prebuiltui";
+import Session from "supertokens-auth-react/recipe/session";
+// highlight-end
+import { useNavigate } from "react-router-dom";
 
 SuperTokens.init({
     appInfo: {
         apiDomain: "...",
         appName: "...",
-        websiteDomain: "...",
+        websiteDomain: "..."
     },
     recipeList: [
         ThirdPartyEmailPassword.init({
             // ...
         }),
+        // ...
     ],
-    //highlight-start
-    // The user will be taken to the custom path when then need to login.
-    getRedirectionURL: async (context) => {
+    // highlight-start
+    async getRedirectionURL(context) {
+        // The user will be taken to this path when they need to login.
         if (context.action === "TO_AUTH") {
-            return "/custom-login-path";
+            return "/auth?showModal=true"; // return the path where you are rendering the Auth UI
         };
+    },
+    // highlight-end
+});
+
+function AuthPopup() {
+  let sessionContext = Session.useSessionContext();
+  const navigate = useNavigate();
+  const urlParams = new URLSearchParams(window.location.search);
+  const [isModalOpen, setIsModalOpen] = useState(urlParams.get("showModal") === "true");
+
+  const openModal = () => setIsModalOpen(true);
+  const closeModal = () => setIsModalOpen(false);
+
+  useEffect(() => {
+    if (sessionContext.loading) return;
+    if (sessionContext.doesSessionExist) {
+      closeModal();
     }
-    //highlight-end
-})
+  }, [sessionContext]);
+
+  if (sessionContext.loading) return null;
+
+  return (
+    <div style={{ textAlign: "center" }}>
+      {sessionContext.doesSessionExist ? (
+        // We wrap this with <SessionAuth /> so that
+        // all claims are validated before showing the logged in UI.
+        // For example, if email verification is switched on, and
+        // the user's email is not verified, then <SessionAuth />
+        // will redirect to the email verification page.
+        <Session.SessionAuth>
+          <h2>You are logged In! </h2>
+          <h3>UserId: {sessionContext.userId}</h3>
+          <button onClick={() => Session.signOut()}>Sign Out</button>
+        </Session.SessionAuth>
+      ) : (
+        <div>
+          <h2>You are logged Out! </h2>
+          <h3>Open the modal below to Sign In</h3>
+          <button onClick={openModal}>Open Modal</button>
+        </div>
+      )}
+
+      <Modal
+        isOpen={isModalOpen}
+        onRequestClose={closeModal}
+        contentLabel="Auth Modal"
+      >
+        {/* highlight-next-line */}
+        <SignInAndUp navigate={navigate} />
+      </Modal>
```

Comment:
if modal is not showing, does this unmount SignInAndUp? Have you actually tried it?

Reply:
Yeah, I have tested it.